web/client/src/library/components/editor/extensions/SQLMeshDialect.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  completeFromList,
+  type CompletionContext,
+  type Completion,
+} from '@codemirror/autocomplete'
+import { keywordCompletionSource, SQLDialect } from '@codemirror/lang-sql'
+import { LanguageSupport } from '@codemirror/language'
+import { type Model } from '~/api/client'
+import { isFalse, isNil } from '~/utils'
+import { sqlglotWorker } from '~/workers'
+
+const cache = new Map<string, (e: MessageEvent) => void>()
+const WHITE_SPACE = ' '
+
+export type ExtensionSQLMeshDialect = (
+  models: Map<string, Model>,
+  options?: { types: string; keywords: string },
+  dialects?: string[],
+) => LanguageSupport
+
+export const SQLMeshDialect: ExtensionSQLMeshDialect = function SQLMeshDialect(
+  models,
+  options = { types: '', keywords: '' },
+  dialects,
+): LanguageSupport {
+  const SQLKeywords = options.keywords + WHITE_SPACE
```

Comment:
Can you please remind me again why all these strings need a trailing white space?

Reply:
in case `options.keywords` coming without trailing space

Reply:
Is it for concatenating with `SQLMeshKeywords` below for `lang`? Does codemirror expect a trailing whitespace? What would happen if `SQLKeywords` and `SQLMeshKeywords` didn't have trailing whitespaces, but instead below is `keywords: (SQLKeywords + WHITE_SPACE + SQLMeshKeywords).toLowerCase()`?